client/game/board/Board.svelte

```diff
@@ -2,29 +2,44 @@
 	export let cards = [];
 
 	import Card from "./Card.svelte";
-	import { createEventDispatcher } from "svelte";
+	import { socket } from "../../connectivity.js";
 	import { areCardsEqual, arrayContainsCard, isValidPlay } from "../shared.js";
 
-	const dispatch = createEventDispatcher();
+	const keyboardMap = ["qwertyu", "asdfghj", "zxcvbnm"].map(line => line.split(""));
+
+	function getKey(i) {
+		if (i >= keyboardMap.length * keyboardMap[0].length) {
+			return "";
+		} else {
+			return keyboardMap[i % 3][parseInt(i / 3)].toUpperCase();
+		}
+	}
 
 	let selection = [];
 
 	function cardClicked(e) {
+		if (selection.length >= 3) {
+			return;
+		}
 		const card = e.detail.card;
 		if (arrayContainsCard(selection, card)) {
 			selection = selection.filter(card => !areCardsEqual(card, e.detail.card));
 		} else {
 			selection = [...selection, card];
 			if (selection.length == 3) {
-				dispatch("play", {
-					selection
-				});
+				socket.emit("play", selection);
 				// submit to server for check and broadcast
-				console.log(isValidPlay(selection));
+				// console.log(isValidPlay(selection));
 				setTimeout(() => (selection = []), 200); // after check
 			}
 		}
 	}
+
+	function handleKeydown(e) {
+		if (e.code == "Backspace") {
```

Comment:
```suggestion
		if (e.code === "Backspace") {
```

Also, maybe include `"Delete"` and `"Escape"`? Your keyboard bindings are on the left side of the keyboard, but the `"Backspace"` button is on the right side of the keyboard. Plus, if it's to clear the selection, I think `"Escape"` and `"Delete"` all make sense here.

Reply:
Added!